pkg/operator/controller/ingress/status.go

```diff
@@ -735,3 +740,37 @@ func computeDNSStatus(ic *operatorv1.IngressController, wildcardRecord *iov1.DNS
 
 	return conditions
 }
+
+// checkZoneInConfig - private utility to check for a zone in the current config
+func checkZoneInConfig(dnsConfig *configv1.DNS, zone configv1.DNSZone) bool {
+
+	// check for public zone ID
+	if dnsConfig.Spec.PublicZone != nil && dnsConfig.Spec.PublicZone.ID != "" && zone.ID != "" {
```

Comment:
Would it simplify the function if before all the checks here you check for `zone.ID = ""` - if it is, then return false early.

Reply:
Yes good point

Reply:
Simplified for PrivateZone check only (as the BZ refers to PrivateZone)